src/main/java/lotto/model/LottoTicketGenerator.java

```diff
@@ -15,9 +17,21 @@ public class LottoTicketGenerator {
                 .collect(Collectors.toList());
     }
 
-    public static LottoTicket generate() {
-        Collections.shuffle(LOTTO_NUMBERS);
-        return LOTTO_NUMBERS.stream()
+    public static LottoTickets generate(LottoPurchaseAmount lottoPurchaseAmount) {
+        int count = (int) lottoPurchaseAmount.getPurchaseAmount() / LottoTicket.PRICE;
+        return new LottoTickets(generate(count));
+    }
+
+    private static List<LottoTicket> generate(int count) {
+        return Stream.generate(LottoTicketGenerator::generate)
+                .limit(count)
+                .collect(Collectors.toList());
+    }
+
+    private static LottoTicket generate() {
+        List<LottoNumber> copiedLottoNumbers = new ArrayList<>(LOTTO_NUMBERS);
+        Collections.shuffle(copiedLottoNumbers);
+        return copiedLottoNumbers.stream()
                 .limit(LottoTicket.SIZE)
                 .collect(Collectors.collectingAndThen(Collectors.toList(), LottoTicket::new));
```

Comment:
Stream을 생성할 필요 없이 바로 `Collection`의 함수인 `subLIst`를 활용해보면 어떨까요? 🙂

```suggestion
        new LottoTicket(copiedLottoNumbers.subList(0, 6))
```